### PR TITLE
feat(providers): add option to apply ratelimiting over single API

### LINKED ISSUE #46: Option to apply ratelimiting over a single API
**Describe the bug**
As of the now, the module provides a way to ratelimit all api except a few that are disabled using a decorator. There should be a way to ratelimit only a few of the many available apis.

**Expected behavior**
There should be a configurable default mode of ratelimiting for a microservice, if it is false, ratelimiting should be disabled for all APIs except those that are enabled using the decorator.

### PR DESCRIPTION
**Pull request by Surbhi-sharma1**
GH-46

## Description

Added option to apply ratelimiting over single API.

Fixes #46 
GH-46
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: src/providers/ratelimit-action.provider.ts [hunk 1]
```diff
@@ -54,6 +54,16 @@ export class RatelimitActionProvider implements Provider<RateLimitAction> {
...
+    if (this.config?.enabledByDefault === true) {
+      await promise;
+    } else if (
+      this.config?.enabledByDefault === false &&
+      metadata &&
+      metadata.enabled
+    ) {
+      await promise;
+    } else {
+      return Promise.resolve();
```
[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply src/providers/ratelimit-action.provider.ts:63]: Done

[the remark this answers is not in the document]
Comment (yeshamavani) [reply src/providers/ratelimit-action.provider.ts:63]: Its for the case when enabledByDefault is set to false and also when decorator is not passed over the API

[the remark this answers is not in the document]
Comment (Surbhi-sharma1) [reply src/providers/ratelimit-action.provider.ts:63]: yes

**Pull request merged**